setup.py

```diff
@@ -73,7 +73,13 @@ def in_virtualenv():
 def get_version():
     cmd = [sys.executable, "tools/read-version"]
     ver = subprocess.check_output(cmd)
-    return ver.decode("utf-8").strip()
+    version = ver.decode("utf-8").strip()
+    # read-version can spit out something like 22.4-15-g7f97aee24
+    # which is invalid under PEP440. If we replace the first - with a +
+    # that should give us a valid version.
+    if "-" in version:
+        version = version.replace("-", "+", 1)
+    return version
```

Comment:
Good add. python sdist should know how to parse our local version output and your "+" aligns with [PEP 440 [+<local version label>]](https://peps.python.org/pep-0440/#local-version-identifiers)